### PR TITLE
MEPTS-457-RM: Change Request in TX TB - Positive Screening Disaggregation

### PR DESCRIPTION
**Pull request by amosh101**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java [hunk 1]
```diff
@@ -541,7 +541,7 @@ public CohortDefinition getPatientsWithAtLeastPosNegInvestigationResultTB() {
             + "    o.voided = 0 AND "
             + "    e.encounter_type IN (${adultoSeguimentoEncounterType},${pediatriaSeguimentoEncounterType})  AND "
             + "    o.concept_id = ${researchResultConcept} AND "
-            + "    o.value_coded IN (${positiveConcept},${negativeConcept}) AND "
+            + "    o.value_coded = ${positiveConcept}AND "
             + "    e.encounter_datetime BETWEEN :startDate AND :endDate AND "
             + "    e.location_id  = :location "
             + "GROUP BY p.patient_id;";
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java:547]: I don't think we need that semi colon(;) inside the string then have another one outside

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java [hunk 2]
```diff
@@ -553,6 +553,75 @@ public CohortDefinition getPatientsWithAtLeastPosNegInvestigationResultTB() {
...
+            + "WHERE "
+            + "    p.voided = 0 AND "
+            + "    e.voided = 0 AND "
+            + "    o.voided = 0 AND "
+            + "    e.encounter_type IN (${adultoSeguimentoEncounterType},${pediatriaSeguimentoEncounterType})  AND "
+            + "    o.concept_id = ${researchResultConcept} AND "
+            + "    o.value_coded = ${negativeConcept} AND "
+            + "    e.encounter_datetime BETWEEN :startDate AND :endDate AND "
+            + "    e.location_id  = :location "
+            + "    GROUP BY p.patient_id) as list;";
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java:615]: same here

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java (hunk 1 above)
Comment (amosh101) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java:547]: Should i remove for all occurences on this file?

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoTrimestralCohortQueries.java [hunk 3]
```diff
@@ -44,15 +44,14 @@ public CohortDefinition getA() {
...
-    cd.addSearch("startedArt", mapStraightThrough(startedArt));
-    cd.addSearch(
-        "transferredIn",
-        map(
-            transferredIn,
-            "onOrAfter=${onOrAfter},onOrBefore=${onOrBefore+12m},location=${location}"));
+    cd.addParameters(getParameters());
+
+    String yearBefore =
+        "onOrAfter=${onOrAfter-12m},onOrBefore=${onOrBefore-12m},location=${location}";
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoTrimestralCohortQueries.java]: As the variable say, yearBefore, to me that being worked from the end of the reporting period, would be 
String yearBefore =
        "onOrAfter=${onOrBefore-12m},onOrBefore=${onOrBefore},location=${location}";

Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoTrimestralCohortQueries.java]: sorry, that is coming from different class that might NOT be related to this PR

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java (hunk 2 above)
Comment (amosh101) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java]: Done

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java (hunk 1 above)
Comment (amosh101) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java]: Done

Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java]: That semi colon inside the sring is NOT required, quickly test after removing those

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java [hunk 4]
```diff
@@ -494,7 +494,7 @@ public CohortDefinition getPatientsStartTBTreatment() {
             + "    o.concept_id = ${tbDrugTreatmentStartDate} AND "
             + "    e.encounter_datetime BETWEEN :startDate AND :endDate AND "
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java:495]: If this method is nolonger applicable, you can consider deleting it

Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java:495]: public CohortDefinition getPatientsStartTBTreatment()

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoTrimestralCohortQueries.java (hunk 3 above)
Comment (amosh101) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoTrimestralCohortQueries.java:50]: Dear @ynurmahomed  please confirm if this now looks as it should.

Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoTrimestralCohortQueries.java:50]: Looks like even this code is NOT supposed to be part of your commits, did you add them? seems it has caused discrepancies in another PR

Comment (ynurmahomed) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoTrimestralCohortQueries.java:50]: This is the way it is supposed to be as per #45. Somehow it got changed after that.

Comment (amosh101) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoTrimestralCohortQueries.java:50]: All alterations to this were added from merging develop, however now @ynurmahomed requested this change was added.

Comment (ynurmahomed) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoTrimestralCohortQueries.java:50]: This changes would work better as part of a separate PR as it is not related to TX_TB and for historical purposes.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java (hunk 4 above)
Comment (amosh101) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java]: Removed

**Pull request merged**